src/libs/ReportUtils.ts

```diff
@@ -762,6 +763,11 @@ Onyx.connect({
                 return;
             }
             handleReportChanged(report);
+            if (report.policyID && report.ownerAccountID === currentUserAccountID && (report.stateNum ?? 0) <= 1) {
+                const policyReports = userReportsOpenOrSubmittedByPolicyId[report.policyID] ?? [];
+                policyReports.push(report);
+                userReportsOpenOrSubmittedByPolicyId[report.policyID] = policyReports;
+            }
```

Comment:
`userReportsOpenOrSubmittedByPolicyId` may hold stale data. e.g. reports that have changed `stateNum` would still exist even though they are neither open nor submitted.

Reply:
Nice catch!